pkgs/by-name/da/davs2/package.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  fetchFromGitHub,
+  gitUpdater,
+  stdenv,
+  testers,
+  buildPackages,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "davs2";
+  version = "1.7";
+
+  src = fetchFromGitHub {
+    owner = "pkuvcl";
+    repo = "davs2";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-SUY3arrVsFecMcbpmQP0+4rtcSRfQc6pzxZDcEuMWPU=";
+  };
+
+  postPatch = ''
+    substituteInPlace ./version.sh \
+      --replace-fail "date" 'date -ud "@$SOURCE_DATE_EPOCH"'
+  '';
+
+  preConfigure = ''
+    # Generate version.h
+    ./version.sh
+    cd build/linux
+    export AS=${lib.getExe buildPackages.nasm}
+  '';
```

Comment:
Nit:

```suggestion
  '';
  
  env.AS = lib.getExe buildPackages.nasm;
```

Reply:
Has the same issue as just setting `env.AS = "nasm";`

Reply:
I'm not sure what you mean. What I proposed here should basically the exact same thing but a little more robustly.

Reply:
It should, but it somehow is not and if I set AS over env it gets ignored and gasm gets used instead, which then fails due to not being nasm.

Reply:
Okay, that's wild. Please add a comment.